src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -42,12 +77,84 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
 
             try
             {
-                // TODO: Implement the actual export logic here.
+                // Get destination type from secret store.
+                DestinationInfo destinationInfo = await GetDestinationInfo(cancellationToken);
+
+                // Connect to the destination using appropriate client.
+                _exportDestinationClient = _exportDestinationClientFactory.Create(destinationInfo.DestinationType);
+
+                await _exportDestinationClient.ConnectAsync(destinationInfo.DestinationConnectionString, cancellationToken);
+
+                // TODO: For now, always restart from the beginning. We will support resume in another work item.
+                exportJobRecord.Progress = new ExportJobProgress(continuationToken: null, page: 0);
```

Comment:
nit: Maybe we can delete this line and simplify the below line to `ExportJobProgress progress = new ExportJobProgress(continuationToken: null, page: 0);`